server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/MailboxGetMethod.scala

```diff
@@ -77,16 +78,15 @@ class MailboxGetMethod @Inject() (serializer: Serializer,
 
   private def getMailboxes(mailboxGetRequest: MailboxGetRequest, mailboxSession: MailboxSession): SFlux[MailboxGetResults] = mailboxGetRequest.ids match {
     case None => getAllMailboxes(mailboxSession).map(MailboxGetResults.found)
-    case Some(ids) =>
-      getAllMailboxes(mailboxSession)
-        .collectSeq()
-        .flatMapMany(mailboxes => SFlux.merge(Seq(
-          SFlux.fromIterable(mailboxes)
-            .filter(mailbox => ids.value.contains(mailbox.id))
-            .map(MailboxGetResults.found),
-          SFlux.fromIterable(ids.value)
-            .filter(id => !mailboxes.map(_.id).contains(id))
-            .map(MailboxGetResults.notFound))))
+    case Some(ids) => SFlux.fromIterable(ids.value)
+      .flatMap(id => mailboxFactory.create(id, mailboxSession)
+        .flatMap { // TODO fixme
```

Comment:
Error message during compilation on IntelliJ:
```
Error:(83, 18) type mismatch;
 found   : Either[Exception,org.apache.james.jmap.mail.Mailbox] => reactor.core.scala.publisher.SMono[_ <: MailboxGetMethod.this.MailboxGetResults]
 required: Either[Exception,org.apache.james.jmap.mail.Mailbox] => reactor.core.scala.publisher.SMono[MailboxGetMethod.this.MailboxGetResults]
        .flatMap({ // TODO fixme
```
Couldn't manage to fix it... Any help appreciated